build.yaml

```diff
@@ -1,21 +1,90 @@
 # vim ft=yaml:fdm=indent
-# To modify an existing job, update the dictionary 'jobs_type'
-# commit, push
-# Then click 'deploy' on the jenkins job related to this repository
-# available on the jenkins homepage.
-
+#
+# build.yaml
+#
+# This file contains the Jenkins job defintions for the Flocker project.
+#
+# We add new or reconfigure existing jobs in Jenkins using the Job DSL plugin.
+# https://github.com/jenkinsci/job-dsl-plugin
+#
+# That plugin consumes 'groovy Job DSL' code through FreeStyle Jenkins job
+# types which contain a 'Process Job DSLs' step.
+#
+# We as part of the provisioning process for the Jenkins Master, configure a
+# job called 'setup_ClusterHQ_Flocker' which is responsible for querying the
+# github clusterhq/flocker respository and retrieving this build.yaml file, as
+# well as the jobs.groovy.j2 jinja file.
+#
+# As part of the provisioning process for the Jenkins Master, we also deploy
+# a small python script '/usr/local/bin/render.py', this script simply contains
+# code to read a YAML file into a dictionary and expand a Jinja2 template using
+# the k,v in that dict.
```

Comment:
I guess this is because we want to write our config in yaml, rather than groovy?

If so, please add something like: "We do this, because we'd rather configure our jobs with YAML than with Groovy."